### DISCUSSION TITLE
stop reporting

### DISCUSSION THREAD

**Discussion by erwinSmith757** [ESPresense/ESPresense#246] [Q&A]
I use it on a D1 mini, but after 24h... it stops finding the beacons,,,, but as soon as i restart the D1 mini... it works flawless again....

wonder if i can create a script in HA, to restart them every few hours

**Comment by Terastar-Paperclip** [comment]
Sorry this sat quiet — a node that works fine and then goes silent until a manual restart sounds like a real stability issue rather than a config one. Flagging this for the core team to take a look at rather than guessing at a fix here; I'll follow up once I hear back.

**Comment by Terastar-Paperclip** [comment]
Tech Lead here. This report is old and sparse (no version/logs), so I can't pin an exact root cause from the thread alone. But "works fine for hours, then silently stops finding beacons until a manual power-cycle" matches a general class of long-uptime BLE/heap issues we're actively chasing right now: #2309 (free heap declines over hours with fingerprint count flat), with an open stream of diagnostic/mitigation PRs (#2051 maxFingerprints cap, #2326 OOM protection for the BLE pipeline, #2471 low-heap guards, #2448/#2313 memory-density work). None of that is merged yet, so there's no fix to hand you today.

The HA-restart-on-a-timer workaround you proposed is a reasonable stopgap in the meantime. If anyone is still seeing this on **current** firmware, please open a fresh bug report with firmware version + logs so we can correlate it with #2309 instead of guessing from a 2022 thread.
